### DISCUSSION TITLE
Working config for an ESP32-S3-MINI-1 + 1.69" 240x280 ST7789V2 TFT display

### DISCUSSION THREAD

**Discussion by andrewchilds** [Bodmer/TFT_eSPI#3256] [Compatible displays and setup files]
Just noting that after a lot of wall-hitting I was able to get this library working with this setup configuration file:

1. ESP32-S3-DevKitM-1 (ESP32-S3-MINI-1)
2. 1.69" 240x280 rounded rectangle TFT display with a ST7789V2 driver (https://www.aliexpress.us/item/3256801802899906.html)

```
// Working configuration for ESP32-S3-MINI-1U chip and a 1.69" 240x280 rounded TFT display with ST7789V2 driver.

#define USER_SETUP_INFO "My_User_Setup"

#define ST7789_2_DRIVER

#define TFT_RGB_ORDER TFT_RGB

#define TFT_WIDTH   240
#define TFT_HEIGHT  280

#define TFT_MOSI    35  // aka SDA
#define TFT_SCLK    36  // aka SCL
#define TFT_CS      26  // Chip select control pin
#define TFT_DC      47  // Data Command control pin
#define TFT_RST     4   // Reset pin (could connect to Arduino RESET pin)
#define TFT_MISO    19  // "For some reason DMA requires that a MISO pin is defined" - https://github.com/Bodmer/TFT_eSPI/discussions/2233

#define LOAD_GLCD     // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2    // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4    // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6    // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7    // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8    // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF    // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define SPI_FREQUENCY         27000000
#define SPI_READ_FREQUENCY    20000000
#define SPI_TOUCH_FREQUENCY   2500000
```

It ended up being that `TFT_MISO` was undefined, and I needed to set `TFT_RGB_ORDER` to `TFT_RGB`. I did not need to define `CGRAM_OFFSET`. Otherwise this uses the default configuration.

Thank you to @Bodmer for creating this library and putting in the time to handle support requests.

**Comment by andrewchilds** [comment]
Documenting here that I was able to shave ~500ms off of `tft.init` by reducing the driver delays in [TFT_Drivers/ST7789_2_Init.h](https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_Drivers/ST7789_2_Init.h), at least for the particular 1.69" 240x280 TFT display I'm using. In my tests this allows the display to appear ~500ms after waking up from deep sleep.

```diff
{
static const uint8_t PROGMEM
  st7789[] = {
    8,
-    TFT_SLPOUT, TFT_INIT_DELAY, 255,
+    TFT_SLPOUT, TFT_INIT_DELAY, 20,
    TFT_COLMOD, 1+TFT_INIT_DELAY, 0x55, 10,
    TFT_MADCTL, 1, 0x00,
    TFT_CASET, 4, 0x00, 0x00, 0x00, 0xF0,
    TFT_PASET, 4, 0x00, 0x00, 0x00, 0xF0,
    TFT_INVON, TFT_INIT_DELAY, 10,
    TFT_NORON, TFT_INIT_DELAY, 10,
-    TFT_DISPON, TFT_INIT_DELAY, 255
+    TFT_DISPON, TFT_INIT_DELAY, 20
    };

    commandList(st7789);
}
```

**Reply by pvetal** [reply]
Hi [andrewchilds](https://github.com/andrewchilds).
How you think, your method can help to "friendship" ESP32 S3 with 4" display ST7796? 
I'm beginner in this, but in another forum peoples wrote something about 8 and 16 bit support.

**Reply by andrewchilds** [reply]
Hi @pvetal, this discussion was just a note for how I was able to get my particular display working. I don't have access to your display and therefore don't know how to get your display working.

**Reply by pvetal** [reply]
Ok, thanks for answer

**Comment by Basitadam** [comment]
> ESP32-S3-MINI-1U chip and a 1.69" 240x280 rounded TFT display with ST7789V2 driver

Hi @andrewchilds. Nice work. Can you send a link to your display? Aliexpress or manufacturer.
Thanks

**Reply by andrewchilds** [reply]
@Basitadam the exact display used in this case is https://www.aliexpress.us/item/3256801802899906.html. I also updated the discussion description with a link.

**Reply by Basitadam** [reply]
@andrewchilds thank you. By the way, according to ST7789V2 datasheet, SPI clock can be as high as 62.5MHz. If needed

**Comment by mjcards** [comment]
Almost exactly what I was looking for as have been hitting walls with a similar waveshare setup. Will try it out and update here. Thanks.
